scalyr_agent/scalyr_logging.py

```diff
@@ -372,6 +372,16 @@ def emit_value(
                 metric_name=metric_name, metric_value=metric_value
             )
 
+        if metric_name in getattr(monitor, "_metric_name_blacklist", []):
```

Comment:
Just for my own education, why use `getattr` here instead of using a `set` to hold the blacklist?

Reply:
That was done to make sure existing tests pass - a lot of them use various mock monitor classes without attributes.

Another approach would be to also update all the affected mock test classes.